caas/kubernetes/provider/bootstrap.go

```diff
@@ -98,6 +104,36 @@ func DecideControllerNamespace(controllerName string) string {
 	return "controller-" + controllerName
 }
 
+// ReScheduler executes a function and keep retrying until timeout if the function returns an expected error.
+func ReScheduler(
```

Comment:
We already have code for this in the juju/retry repo, eg

```
	err := retry.Call(retry.CallArgs{
		Attempts:    30,
		Delay:       time.Second,
		MaxDelay:    time.Minute, // because 2**29 seconds is beyond reasonable
		BackoffFunc: retry.DoubleDelay,
		Clock:       clock,
		Func: func() error {
			_, err := inst.DeleteSecurityGroup(group)
			if err == nil || isNotFoundError(err) {
				logger.Debugf("deleting security group %q", group.Name)
				return nil
			}
			return errors.Trace(maybeConvertCredentialError(err, ctx))
		},
		IsFatalError: func(err error) bool {
			return common.IsCredentialNotValid(err)
		},
		NotifyFunc: func(err error, attempt int) {
			logger.Debugf("deleting security group %q, attempt %d", group.Name, attempt)
		},
	})
```

Reply:
got it, changed to use this